app/src/main/java/com/github/fribourgsdp/radio/Song.kt

```diff
@@ -1,17 +1,24 @@
 package com.github.fribourgsdp.radio
 
 import java.util.*
+import java.util.concurrent.CompletableFuture
 
-class Song (songName: String, artistName: String, songLyrics: String) {
+class Song (songName: String, artistName: String, private val songLyrics: CompletableFuture<String>) {
```

Comment:
Wouldn't be a way to force the completion of the songLyrics in the constructor so that we don't have to play around with a CompletableFuture? Like doing what to do in line 15 in the constructor directly?
edit: typo

Reply:
I thought doing it this way would reduce the waiting time, delaying the get() to the point where we really need it. We don't "play around", given that it is private and that when we get this property, a call to get() is issued.